cpmpy/expressions/globalconstraints.py

```diff
@@ -209,7 +209,7 @@ def __init__(self, *args):
     def decompose(self):
         """Returns the decomposition
         """
-        return [var1 == var2 for var1, var2 in all_pairs(self.args)]
+        return [var1 == var2 for var1, var2 in zip(self.args[:-1], self.args[1:])]
```

Comment:
this is cryptic... can you add a comment about what you do (sufficient to post on pairs of forward/backward ordering of the array) and why (why? if this is true, then just the subsequent pairs in the sequence is also sufficient; but why not just post all?)

Reply:
I thought this would be more efficient as the number of constraints we have to generate is now `n` instead of `n^2`.
Some documentation might indeed be useful tho...